### PR TITLE
disk acct idx on by default on validators and ledger-tool

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem

cancel with --disable-disk-accounts-index
override size with --accounts-index-memory-limit-mb

#24251
#### Summary of Changes

Fixes #

### CODE REVIEW

**Comment by t-nelson** [comment]
> Do we need a way on cli to disable disk acct idx? There currently isn't one.

Yeah I'd prefer a (hidden at first) CLI arg rather than forcing it on like this. Start with hidden `--enable-disk-accounts-index`, when we're ready, deprecate and add a `--disable-disk-accounts-index` with inverted logic throughout

**Comment by jeffwashington** [comment]
will try again later

**Comment by t-nelson** [comment]
we try not to gate on `ClusterType` anymore (pre-feature program kludge).  Can we commit to default disk-index enabled by the time MB goes to 1.10 (optimistically ~2wks)?

File: ledger-tool/src/main.rs [hunk 1]
```diff
@@ -2065,6 +2065,9 @@ fn main() {
                     accounts_index_config.index_limit_mb = Some(limit);
                 } else if arg_matches.is_present("disable_accounts_disk_index") {
                     accounts_index_config.index_limit_mb = None;
+                } else {
+                    // disk acct idx on by default
+                    accounts_index_config.index_limit_mb = Some(10_000);
```
Comment (t-nelson) [inline ledger-tool/src/main.rs:2070]: Is there a good place that we can name `10_000` something like `DEFAULT_ACCOUNTS_INDEX_LIMIT_MB` ?

**Review by t-nelson — APPROVED** [review]
much better!  just one nit

**Comment by jeffwashington** [comment]
moved to #24251

**Pull request closed without merging**